tests/testthat/test-shinytest2_filteredData_DF.R

```diff
@@ -0,0 +1,173 @@
+      teal_slice(dataname = "mtcars", id = "4 cyl", title = "4 Cylinders", expr = "cyl == 4"),
+      teal_slice(dataname = "mtcars", varname = "mpg", selected = c(20.0, 25.0), keep_na = FALSE, keep_inf = FALSE),
+      include_varnames = list(iris = c("Species", "Sepal.Length")),
+      exclude_varnames = list(mtcars = "cyl"),
+      count_type = "all",
+      allow_add = TRUE
+    )
+  )
+
+  ui <- fluidPage(
+    fluidRow(
+      column(
+        width = 9,
+        tabsetPanel(
+          tabPanel(title = "iris", dataTableOutput("iris_table")),
+          tabPanel(title = "mtcars", dataTableOutput("mtcars_table"))
+        )
+      ),
+      # ui for the filter panel
+      column(width = 3,
+                    # What we want to test:
+                    datasets$ui_filter_panel("filter_panel"))
+    )
+  )
+
+  server <- function(input, output, session) {
+    # this is the shiny server function for the filter panel and the datasets
+    # object can now be used inside the application
+    datasets$srv_filter_panel("filter_panel")
+
+    # get the filtered datasets and put them inside reactives for analysis
+    iris_filtered_data <- reactive(datasets$get_data(dataname = "iris", filtered = TRUE))
+    mtcars_filtered_data <- reactive(datasets$get_data(dataname = "mtcars", filtered = TRUE))
+
+    output$iris_table <- renderDataTable(iris_filtered_data())
+    output$mtcars_table <- renderDataTable(mtcars_filtered_data())
+  }
+
+  app <- shinyApp(ui, server)
+  app_driver <- shinytest2::AppDriver$new(
+    app,
+    name = name,
+    variant = variant,
+    timeout = default_idle_timeout,
+    load_timeout = default_idle_timeout,
+    wait = TRUE,
+    seed = 20250626
+  )
+  app_driver
+}
+
+testthat::test_that("initializes", {
+  app_driver <- app()
+  testthat::expect_true(is_visible(app_driver, "#filter_panel"))
+  testthat::expect_true(is_visible(app_driver, "#filter_panel-overview-main_filter_accordion"))
+  testthat::expect_equal(app_driver$get_text("#filter_panel-overview-table > table > tbody > tr:nth-child(1) > td:nth-child(1)"),
+                         "iris")
+  testthat::expect_equal(app_driver$get_text("#filter_panel-overview-table > table > tbody > tr:nth-child(2) > td:nth-child(1)"),
+                         "mtcars")
+
+  testthat::expect_true(is_visible(app_driver, "#filter_panel-active"))
+  testthat::expect_true(is_visible(app_driver, "#filter_panel-active-iris-dataset_filter_accordion"))
+  testthat::expect_true(is_visible(app_driver, "#filter_panel-active-mtcars-dataset_filter_accordion"))
+
+})
+
+testthat::test_that("Toggle visibility of Active Filter Summary", {
+  app_driver <- app()
+
+  selector_collapsable <- get_class(".accordion-item:nth-of-type(1) > div.accordion-collapse")
+  out <- app_driver$get_js(selector_collapsable)
+  testthat::expect_length(out, 4)
+  testthat::expect_true("show" %in% unlist(out[[1]]))
```

Comment:
This is repeated 3 times during tests, can we just create a test that checks if this is the default?